wolfcrypt/src/ecc.c

```diff
@@ -11264,16 +11284,33 @@ int wc_ecc_encrypt(ecc_key* privKey, ecc_key* pubKey, const byte* msg,
     if ( (msgSz%blockSz) != 0)
         return BAD_PADDING_E;
 
+#ifdef WOLFSSL_ECIES_OLD
     if (*outSz < (msgSz + digestSz))
         return BUFFER_E;
+#else
+    if (*outSz < (pubKeySz + msgSz + digestSz))
+        return BUFFER_E;
+#endif
 
 #ifdef ECC_TIMING_RESISTANT
     if (ctx->rng != NULL && privKey->rng == NULL)
         privKey->rng = ctx->rng;
 #endif
 
+#ifndef WOLFSSL_ECIES_OLD
+    if (privKey->type == ECC_PRIVATEKEY_ONLY) {
+        ret = wc_ecc_make_pub_ex(privKey, NULL, privKey->rng);
+        if (ret != 0)
+            return ret;
+    }
+    ret = wc_ecc_export_x963(privKey, out, &pubKeySz);
+    if (ret != 0)
+        return ret;
+    out += pubKeySz;
+#endif
+
 #ifdef WOLFSSL_SMALL_STACK
-    sharedSecret = (byte*)XMALLOC(ECC_MAXSIZE, NULL, DYNAMIC_TYPE_ECC_BUFFER);
+    sharedSecret = (byte*)XMALLOC(sharedSz, NULL, DYNAMIC_TYPE_ECC_BUFFER);
```

Comment:
Heap hint `ctx->heap`. I know static memory doesn't support small stack, but there are other uses of the heap hint besides static memory. Can you also add this to the other XMALLOC/XFREE places below?

Reply:
Fixed